clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,921 @@
+package v1repl
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"text/tabwriter"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/fatih/color"
+	"github.com/influxdata/go-prompt"
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+)
+
+type Client struct {
+	clients.CLI
+	PersistentQueryParams
+	api.LegacyQueryApi
+	api.PingApi
+	api.OrganizationsApi
+	api.WriteApi
+	api.DBRPsApi
+}
+
+type PersistentQueryParams struct {
+	clients.OrgParams
+	Database        string
+	Password        string
+	Username        string
+	RetentionPolicy string
+	Precision       string
+	Format          FormatType
+	Pretty          bool
+	historyFilePath string
+	// Autocompletion Storage
+	Databases         []string
+	RetentionPolicies []string
+	Measurements      []string
+}
+
+func (c *Client) clear(cmd string) {
+	args := strings.Split(strings.TrimSuffix(strings.TrimSpace(cmd), ";"), " ")
+	v := strings.ToLower(strings.Join(args[1:], " "))
+	switch v {
+	case "database", "db":
+		c.Database = ""
+		c.RetentionPolicies = []string{}
+		fmt.Println("database context cleared")
+		return
+	case "retention policy", "rp":
+		c.RetentionPolicy = ""
+		fmt.Println("retention policy context cleared")
+		return
+	default:
+		if len(args) > 1 {
+			fmt.Printf("invalid command %q.\n", v)
+		}
+		fmt.Println(`Possible commands for 'clear' are:
+    # Clear the database context
+    clear database
+    clear db
+
+    # Clear the retention policy context
+    clear retention policy
+    clear rp
+		`)
+	}
+}
+
+func DefaultPersistentQueryParams() PersistentQueryParams {
+	return PersistentQueryParams{
+		Format:    CsvFormat,
+		Precision: "ns",
+	}
+}
+
+func (c *Client) readHistory() []string {
+	// Only load/write history if HOME environment variable is set.
+	var historyDir string
```

Comment:
This repeated code should be extracted - I think we should calculate the history directory at startup similar to the old CLI code